### ISSUE TITLE
Holding down spacebar leads to excessive nodes

### ISSUE THREAD

**Issue by bborkmiller** [openstreetmap/iD#3633]
Being able to hold down the spacebar to trace a river/lake/road is a great feature, but to my eyes this method creates a lot of unnecessary nodes. See [Lane Lake](http://www.openstreetmap.org/way/457950366) for an example of object that exhibits this problem. (I traced it partly by holding down spacebar and partly by clicking.)

From #2784 it looks like iD makes sure the nodes are a certain minimum distance apart. Would it be difficult for iD to not drop a node unless the new segment resulted in some minimum curvature? Ideally, if you traced a straight road (and had a very steady mouse hand) it would drop a node at the beginning and one at the end, and none in between.

Alternatively, a simplify operation (#3116) would help with this. Or maybe a simplification algorithm could be run on the new nodes when the user lets up on the spacebar? Or deselects the line?

**Comment by bhousel** [comment]
Thanks @bborkmiller, I agree this would be a nice-to-have feature, to have iD automatically remove unnecessary points while in draw mode.  #3116 would take care of this too.

**Comment by GoodClover** [comment]
I think this is a necessary addition, as every now and again someone pops up "discovering" this and people create lakes etc with excessive amounts of nodes.

Example:
![image](https://user-images.githubusercontent.com/27222523/102900843-c821aa80-4464-11eb-9577-c824e75feb7a.png)

**Comment by quincylvania** [comment]
@GoodClover Yep, this is still something we'd like to solve. Though I'd say it isn't the highest priority since excessive nodes like this aren't all that common and it's usually more an annoyance than a big problem.
